Model/Adminhtml/Source/SOCSelect.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * 2018 Alma / Nabla SAS
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma / Nabla SAS <contact@getalma.eu>
+ * @copyright 2018 Alma / Nabla SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ *
+ */
+
+namespace Alma\MonthlyPayments\Model\Adminhtml\Source;
+
+use Alma\MonthlyPayments\Helpers\ShareOfCheckout\ShareOfCheckoutHelper;
+use Magento\Framework\Option\ArrayInterface;
+
+/**
+ * Class APIModes
+ */
+class SOCSelect implements ArrayInterface
+{
+
+    /**
+     * @var ShareOfCheckoutHelper
+     */
+    private $shareOfCheckoutHelper;
+
+    /**
+     * @param ShareOfCheckoutHelper $shareOfCheckoutHelper
+     */
+    public function __construct(
+        ShareOfCheckoutHelper $shareOfCheckoutHelper
+    ) {
+        $this->shareOfCheckoutHelper = $shareOfCheckoutHelper;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toOptionArray(): array
+    {
+        $arrayResult = [];
+        if ($this->shareOfCheckoutHelper->getShareOfCheckoutSelectorValue() == 2) {
```

Comment:
this test is the same than the test in `SocMessage::isDisplayed`.

Maybe your `ShareOfCheckoutHelper` is the best place to store the constants for `0`, `1` & `2` AND/OR add a public method to test if the `selector === 2`.

For example:

```
constant SELECTOR_NO  = 0;
constant SELECTOR_YES = 1;
constant SELECTOR_NAN = 2;
public function isYesOrNoSelected() {
    $this-> getShareOfCheckoutSelectorValue !== self::SELECTOR_NAN;
}
```

And this will replace your current test by:

```
        if ( ! $this->shareOfCheckoutHelper->isYesOrNoSelected ) {
```